airflow/decorators/base.py

```diff
@@ -457,6 +457,11 @@ def _expand(self, expand_input: ExpandInput, *, strict: bool) -> XComArg:
         end_date = timezone.convert_to_utc(partial_kwargs.pop("end_date", None))
         if partial_kwargs.get("pool") is None:
             partial_kwargs["pool"] = Pool.DEFAULT_POOL_NAME
+        if partial_kwargs.get("pool_slots") and partial_kwargs["pool_slots"] < 1:
+            dag_str = ""
+            if partial_kwargs.get("dag"):
+                dag_str = f" in dag {partial_kwargs['dag'].dag_id}"
```

Comment:
`dag` is already available in the local scope, why do you need to get it from `partial_kwargs`?

Reply:
I will make this change.

Reply:
I have used the dag variable from the local scope in my latest commit.